adpcm-decoder/main.c

```diff
@@ -14,14 +15,34 @@ int main(int argc, char* argv[]) {
 
   FILE* in = fopen(argv[1], "rb");
 
-  // Read wave header
+  char riff_chunk_id[4];
+  fread(riff_chunk_id, 4, 1, in);
+  if (strncmp(riff_chunk_id, "RIFF", 4) != 0) {
+    fprintf(stderr, "Unexpected chunk: '%.4s'. Expected 'RIFF'\n", riff_chunk_id);
+    return 1;
+  }
 
-  // FIXME: Don't skip here, progress here
-  // 12 = chunk type ('fmt ')
-  // 16 = chunk  size (uint32_t chunk_size = 20)
-  fseek(in, 20, SEEK_SET);
+  uint32_t riff_chunk_size;
+  fread(&riff_chunk_size, 4, 1, in);
+  printf("RIFF size: %d\n", riff_chunk_size);
 
-  // Parse 'fmt ' chunk
+  char wave_chunk_id[4];
+  fread(wave_chunk_id, 4, 1, in);
+  if (strncmp(wave_chunk_id, "WAVE", 4) != 0) {
+    fprintf(stderr, "Unexpected chunk: '%.4s'. Expected 'WAVE'\n", wave_chunk_id);
+    return 1;
+  }
+
+  char fmt_chunk_id[4];
+  fread(fmt_chunk_id, 4, 1, in);
+  if (strncmp(fmt_chunk_id, "fmt ", 4) != 0) {
+    fprintf(stderr, "Unexpected chunk: '%.4s'. Expected 'fmt '\n", fmt_chunk_id);
+    return 1;
+  }
+
+  uint32_t fmt_chunk_size;
+  fread(&fmt_chunk_size, 4, 1, in);
+  printf("Chunk size: %d\n", fmt_chunk_size);
```

Comment:
This is still inconsistent. The others are "RIFF size" and "Data size"; so why isn't this one "fmt size" or "Format size"?